design/specs/listing-views.md

```diff
@@ -1,27 +1,120 @@
+
+#### Scenario 1b: From the schema explorer sidebar
+
+- User opens a schema and sees the [schema explorer sidebar](#schema-explorer-sidebar).
+- User sees a list of all tables and views for the current schema.
+  - The list shows both object types by default but can be filtered to only show tables or views using the [sidebar tabs](#sidebar-tabs) at the top of the sidebar.
+
+### Scenario 2: User encounters an error in loading a list of views for a currently open schema
+
+#### Scenario 2a: From the schema details view
+
+- User opens a schema and sees the [schema details page](#schema-details-page).
+The list of objects for the schema fails to load.
+- Under the schema name, an error message is displayed indicating that the loading of objects failed.
+- The error notice contains details about the error and actions, if available, that the user can perform to correct the problem.
+
+#### Scenario 2b: From the schema explorer sidebar
+
+- User opens a schema and sees the [schema explorer sidebar](#schema-explorer-sidebar).
+The list of objects for the schema fails to load.
+- An error message is displayed indicating to the user that the loading of objects failed.
+- The error notice contains details about the error and actions, if available, that the user can perform to correct the problem.
+
+### Scenario 3: User encounters an error while opening a view in a currently open schema
+
+#### Scenario 3a: From the schema explorer sidebar
+
+- The user opens the schema explorer by navigating to it via the sidebar or by URL pointing to that object.
+- The object fails to load, and the system displays an error notice.
+
+#### Scenario 3b: From an open tab
+
+- The user navigates to a tab, which represents a previously open view.
+- The object fails to load, and the system displays an error notice.
+
+### Scenario 4: User searches a view by name
+
+#### Scenario 4a: From the schema explorer view's sidebar
+
+- The user opens the schema explorer by navigating to it via the sidebar or by URL pointing to that object.
+- The user enters a search term in the search input box located at the top of the sidebar list.
+- The list is filtered to show objects whose name matches the search term.
+- The list contains metadata associated with the objects like record and column count.
+  - If no results are found, the system displays a notice to indicate the lack of results and potential causes.
+- Users can clear the search by clearing the input box content or clicking on the `Clear Search` link displayed along with the results summary.
+
+## Views
+
+### Schema Details Page
+
+The schema details page will be shown when a schema is open and notables are selected. It will contain the schema's name, the total number of tables and views, and additional sections that we can use to present the user with relevant information or links. Examples of content that could go into this section are:
+
+- Recently opened views or tables
```

Comment:
Can you add a note here that these are examples and not meant to be part of the implementation of this spec.

Reply:
If any of these are not examples and should be implemented (maybe recently opened?), please indicate those.